core_contracts/loans/loans/loans.py

```diff
@@ -469,12 +484,13 @@ def getDataBatch(self, _name: str, _snapshot_id: int,
     @external
     def checkForNewDay(self) -> (int, bool):
         day = self.getDay()
-        new_day: bool = False
-        if day > self._current_day.get():
+        new_day = False
+        current_day = self._current_day.get()
+        if current_day < day <= self._continuous_reward_day.get():
             new_day = True
             self._current_day.set(day)
             self._positions._take_snapshot()
-            self.check_dead_markets()
+        self.check_dead_markets()
```

Comment:
Calling check_dead_markets() every time checkForNewDay() is called may be too often. Previously it was only called once per day. is_dead() makes three interscore calls - https://github.com/balancednetwork/balanced-contracts/blob/continuous-integration/core_contracts/loans/loans/assets.py#L140

Reply:
Is it okay or should we optimize it?